providers/provider.go

```diff
@@ -16,11 +16,11 @@ import (
 type Provider interface {
```

Comment:
To your point earlier, do we want to rerun the package too?

Reply:
Sure. I will rename to metric_providers